R/call_map_api

```diff
@@ -0,0 +1,454 @@
+#' "TAXON"), leave this empty and get results with the standard settings first.
+#' In this case, the function willtry to generate a suitable exponential
+#' sequence of break values estimated from the total number of species
+#' occurrence records available for this key.
+#' (\code{default = <dynamic>})
+#'
+#' @param nbreaks Numeric value to specify the number of breaks to
+#' use for the binning of raster values. Should be between 10 and 50. If breaks
+#' are specified this values will be ignored.
+#' (\code{default = 50})
+#'
+#' @param x Numeric value to specify the left binding longitude for horizontal
+#' position of the raster extent.
+#' (\code{default = 0})
+#'
+#' @param y Numeric value to specify the top binding latitude for vertical
+#' position of the raster extent.
+#' (\code{default = 0})
+#'
+#' @param z Numeric value to specify the zoom factor of the raster extent.
+#' (\code{default = 0})
+#'
+#' @param crs_string String to specify the projection of the output raster.
+#' (\code{default = '+proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0'})
+#'
+#' @param map_api_path String to indicate the address of the GBIF map api
+#' endpoint. This is included to allow for manual adjustment should the map api
+#' ever be changed by GBIF. The default value should be used!
+#' (\code{default = '/v1/map/density/tile'})
+#'
+#' @return A list with two objects: [[1]] $raster: An object of the class
+#' \code{\link[raster]{raster}} and [[2]] $url: a string containing the URL generated
+#' for the map API call. The raster package should be installed and loaded to
+#' handle the output of the function. Raster values represent the number of
+#' geo-referenced occurrence records in each of the cells. Notice that values
+#' indicate integer ranges ("bins") rather than exact values. The bins can be
+#' adjusted using the \code{breaks} parameter.
+#'
+#' @details This function uses the arguments passed on to generate a query
+#' to the GBIF web map API. The API returns a web tile object that can be read
+#' as a png and converted into an R raster object. The break values or nbreaks
+#' generate a custom colour palette for the web tile, with each bin
+#' corresponding to one grey value. After retrieval, the raster is reclassified
+#' to the actual break values. This is a somewhat hacky but nonetheless
+#' functional solution in the absence of a GBIF raster API implementation.
+#'
+#' @author Jan Laurens Geffert, \email{laurensgeffert@@gmail.com}
+#' @references \url{http://www.gbif.org/developer/maps}
+#' @keywords web map tile, raster, GBIF, bias grid
+#' @examples \dontrun{
+#'
+#' # Make sure required package is loaded
+#' library(raster)
+#'
+#' # To get a map of all bird records on GBIF:
+#' # First find the key for birds on GBIF backbone taxonomy
+#' BirdBackboneKey <- name_backbone('aves')$classKey
+#' # Then run the function with standard settings
+#' BirdDiversityMap <- map_fetch(key = BirdBackboneKey)
+#' plot(BirdDiversityMap)
```

Comment:
No, your function output has two named elements, and you have to pass the `raster` object to `plot()`